src/app/services/api.service.ts

```diff
@@ -16,12 +16,12 @@ export class ApiService {
     return await data.json() ?? [];
   }
 
-  async getTaskById(id:number): Promise<CardComponent | undefined> {
+  async getTaskById(id: number): Promise<Task> {
```

Comment:
Where are you using this method? 
As we do not see where it might be used at first sight.

Reply:
We do not use it, we created it just in case we may use it in the future

Reply:
That shouldn't be in the code then